src/hardware/serialport/softmodem.cpp

```diff
@@ -355,6 +347,44 @@ char CSerialModem::GetChar(char * & scan) const {
 	return ch;
 }
 
+void CSerialModem::SetModemSpeed(const uint32_t cfg_val) {
+	modem_bps_config   = cfg_val;
+	const auto reported_bps = UpdateConnectString();
+
+	if (modem_bps_config) {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will report connection speed "
+		        "of %d bits per second",
+		        GetPortNumber(),
+		        reported_bps);
+	} else {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will synchronize reported speed "
+		        "to port baud rate",
+		        GetPortNumber());
+	}
+}
+
+uint32_t CSerialModem::UpdateConnectString() {
+	uint32_t connect_val;
```

Comment:
Initialize it to zero (or rework the body to make it a const and only set it once).